lib/ui/semantics/attributed_string.cc

```diff
@@ -0,0 +1,107 @@
+  Dart_IntegerToInt64(dart_start, &start);
+  Dart_IntegerToInt64(dart_end, &end);
+
+  // Gets the type from the dart_string_attribute.
+  flutter::StringAttributeType type;
+  auto dart_type =
+      Dart_GetField(dart_string_attribute, dart_string_attribute_type_string);
+  auto dart_type_index =
+      Dart_GetField(dart_type, dart_string_attribute_type_index_string);
+  int64_t type_int;
+  Dart_IntegerToInt64(dart_type_index, &type_int);
+  type = static_cast<flutter::StringAttributeType>(type_int);
+
+  switch (type) {
+    case flutter::StringAttributeType::kSpellOut: {
+      std::shared_ptr<flutter::SpellOutStringAttribute> attribute =
+          std::make_shared<flutter::SpellOutStringAttribute>();
+      attribute->start = start;
+      attribute->end = end;
+      attribute->type = type;
+      return attribute;
+    }
+    case flutter::StringAttributeType::kLocale: {
+      // Gets the locale string from the dart_string_attribute;
+      auto dart_string_attribute_args_string =
+          Dart_NewStringFromCString(flutter::kStringAttributeArgsFieldName);
+      auto dart_string_attribute_args = Dart_GetField(
+          dart_string_attribute, dart_string_attribute_args_string);
+      const char* locale_string = NULL;
+      Dart_StringToCString(dart_string_attribute_args, &locale_string);
+      std::shared_ptr<flutter::LocaleStringAttribute> attribute =
+          std::make_shared<flutter::LocaleStringAttribute>();
+      attribute->start = start;
+      attribute->end = end;
+      attribute->type = type;
+      attribute->locale = locale_string;
+      return attribute;
+    }
+  }
+}
+
+flutter::AttributedString DartConverter<
+    flutter::AttributedString>::FromArguments(Dart_NativeArguments args,
+                                              int index,
+                                              Dart_Handle& exception) {
+  auto dart_attributed_string = Dart_GetNativeArgument(args, index);
+  FML_DCHECK(!LogIfError(dart_attributed_string));
+
+  // Declare field name for AttributedString
+  auto dart_attributed_string_string_string =
+      Dart_NewStringFromCString(flutter::kAttributedStringStringFieldName);
+  auto dart_attributed_string_attributes_string =
+      Dart_NewStringFromCString(flutter::kAttributedStringAttributesFieldName);
+
+  // Gets the string from the dart_attributed_string.
+  const char* string = NULL;
+  auto dart_attributed_string_string = Dart_GetField(
+      dart_attributed_string, dart_attributed_string_string_string);
+  Dart_StringToCString(dart_attributed_string_string, &string);
+
+  // Gets the attributes from the dart_attributed_string.
+  auto dart_attributed_string_attributes = Dart_GetField(
+      dart_attributed_string, dart_attributed_string_attributes_string);
+  flutter::StringAttributes attributes =
+      DartConverter<flutter::StringAttributes>::FromDart(
+          dart_attributed_string_attributes);
+  return {.string = string, .attributes = attributes};
```

Comment:
I have two concerns about converting this to NativeFieldWrapper2. The first one is the web implementation will be completely different from the mobile. This may make it hard to maintain. The other i am not sure how inheritance would work for NativeFieldWrapper2 since the StringAttribute will have subclasses.

I think I will just use TypeData array for now.